src/Allocator/Process.hs

```diff
@@ -0,0 +1,27 @@
+module Allocator.Process where
+
+import Foundation
+import Manager
+import Control.Effect.State
+
+newtype Pid = Pid Int deriving Eq
+newtype Mem = Mem [Page 'Ram] deriving Eq
+data Process = Process { pid :: Pid } deriving Eq
+
+
+-- list of all processes
+type PTable = [Process]
```

Comment:
Do we really need this? Can't we use the same Map, that we're using to store Pages for different Processes? Process is just a wrapper around Pid, which is a wrapper around Int, it would be OK to do this, I think. Then we would have fewer abstractions, but the same expressive power...

Reply:
The problem is that searching through map is much slower than searching through list. Though having a duplicate structure isn't a solution, I just leaved it as a reminder about this problem.

We could come up with how to generate and return  `ProcessId` (maybe just a list generator of pid-s and tests without implementing a client) to client after creation of process and then implicitly or explicitly pass it to allocator during allocation of page, such way that allocator could just retrieve pages by pid as a value by key. And that would be sufficient enough, I guess